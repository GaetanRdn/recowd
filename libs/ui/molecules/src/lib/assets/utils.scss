@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
