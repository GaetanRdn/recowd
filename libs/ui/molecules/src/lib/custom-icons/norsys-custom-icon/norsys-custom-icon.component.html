<rc-custom-icon [name]="name" [type]="type" [width]="width"></rc-custom-icon>
