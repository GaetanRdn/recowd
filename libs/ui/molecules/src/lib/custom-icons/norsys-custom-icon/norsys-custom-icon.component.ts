import { ChangeDetectionStrategy, Component } from '@angular/core';
import { CustomIconComponent } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-norsys-custom-icon',
  standalone: true,
  imports: [CustomIconComponent],
  templateUrl: './norsys-custom-icon.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class NorsysCustomIconComponent extends CustomIconComponent {
  public override readonly name = 'assets/images/norsys';
}
