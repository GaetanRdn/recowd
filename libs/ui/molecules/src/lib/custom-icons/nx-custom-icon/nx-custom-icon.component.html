<rc-custom-icon [name]="name"></rc-custom-icon>
