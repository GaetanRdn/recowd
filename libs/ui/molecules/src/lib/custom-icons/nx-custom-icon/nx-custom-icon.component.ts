import { Component } from '@angular/core';
import { CustomIconComponent } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-nx-custom-icon',
  standalone: true,
  imports: [CustomIconComponent],
  templateUrl: './nx-custom-icon.component.html',
})
export class NxCustomIconComponent extends CustomIconComponent {
  public override readonly name = 'assets/images/nx';
}
