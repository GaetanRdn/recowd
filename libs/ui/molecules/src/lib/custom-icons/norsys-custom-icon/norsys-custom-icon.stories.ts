import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { NorsysCustomIconComponent } from './norsys-custom-icon.component';

export default {
  title: 'molecules/custom-icons/norsys-custom-icon',
  parameters: {
    jest: 'norsys-custom-icon',
  },
  decorators: [
    moduleMetadata({
      imports: [NorsysCustomIconComponent],
    }),
  ],
} as Meta<NorsysCustomIconComponent>;

const Template: Story<NorsysCustomIconComponent> = () => ({
  template: `<rc-norsys-custom-icon></rc-norsys-custom-icon>`,
});

export const Default = Template.bind({});
