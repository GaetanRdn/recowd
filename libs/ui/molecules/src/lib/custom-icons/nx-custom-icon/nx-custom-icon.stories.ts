import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { NxCustomIconComponent } from './nx-custom-icon.component';

export default {
  title: 'molecules/custom-icons/nx-custom-icon',
  parameters: {
    jest: 'nx-custom-icon',
  },
  decorators: [
    moduleMetadata({
      imports: [NxCustomIconComponent],
    }),
  ],
} as Meta<NxCustomIconComponent>;

const Template: Story<NxCustomIconComponent> = () => ({
  template: `<rc-nx-custom-icon></rc-nx-custom-icon>`,
});

export const Default = Template.bind({});
