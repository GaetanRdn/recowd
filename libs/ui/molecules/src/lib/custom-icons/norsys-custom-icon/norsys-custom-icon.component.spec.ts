import { TestBed, waitForAsync } from '@angular/core/testing';

import { NorsysCustomIconComponent } from './norsys-custom-icon.component';
import { TemplateLookup } from '@recowd/test/utils';
import { NxCustomIconComponent } from '../nx-custom-icon/nx-custom-icon.component';
import { Component } from '@angular/core';

describe('NorsysCustomIconComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [NorsysCustomIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
  }));

  it.each`
    width | height
    ${32} | ${undefined}
    ${16} | ${24}
  `(
    'should create with name $name, width $width and height $height',
    ({ width, height }) => {
      templateLookup.hostComponent.width = width;
      templateLookup.hostComponent.height = height;

      templateLookup.detectChanges();

      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }
  );
});

@Component({
  template: ` <rc-norsys-custom-icon
    [width]="width"
    [height]="height"
  ></rc-norsys-custom-icon>`,
})
class HostComponent {
  public width!: NxCustomIconComponent['width'];
  public height!: NxCustomIconComponent['height'];
}
