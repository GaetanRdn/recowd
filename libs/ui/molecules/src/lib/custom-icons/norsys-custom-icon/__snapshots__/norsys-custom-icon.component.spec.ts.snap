// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NorsysCustomIconComponent should create with name $name, width 16 and height 24 1`] = `
<rc-norsys-custom-icon>
  <rc-custom-icon>
    <object
      data="assets/images/norsys.svg"
      height="24"
      type="image/svg+xml"
      width="24"
    >
      <img
        alt="assets/images/norsys"
        src="assets/images/norsys.png"
      />
    </object>
  </rc-custom-icon>
</rc-norsys-custom-icon>
`;

exports[`NorsysCustomIconComponent should create with name $name, width 32 and height undefined 1`] = `
<rc-norsys-custom-icon>
  <rc-custom-icon>
    <object
      data="assets/images/norsys.svg"
      height="24"
      type="image/svg+xml"
      width="24"
    >
      <img
        alt="assets/images/norsys"
        src="assets/images/norsys.png"
      />
    </object>
  </rc-custom-icon>
</rc-norsys-custom-icon>
`;
